src/main/java/com/guysfromusa/carsgame/control/CollisionMonitor.java

```diff
@@ -0,0 +1,79 @@
+package com.guysfromusa.carsgame.control;
+
+import com.guysfromusa.carsgame.v1.model.Car;
+import com.guysfromusa.carsgame.v1.model.Point;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Created by Konrad Rys, 03.05.2018
+ */
+@Component
+public class CollisionMonitor {
+
+    public void execute(List<Car> cars){
```

Comment:
the idea is that we calculate destinations and then check collisions for all the cars on the map, right?

Reply:
will it also accept all walls -> seems to work as well

Reply:
mutating input params is nasty (not only in functional programing), instead we could maybe return list of crashed cars/carIds